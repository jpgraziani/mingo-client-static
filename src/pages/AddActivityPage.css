.add-activity-container {
  background-color: var(--deep-rose);
  width: 100vw;
}

.activity-form {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  margin: 0 auto;
  justify-content: center;
  align-items: center;
  padding: 4rem 0;
  color: var(--white);
}

.activity-form hr {
  margin: 0;
}

.add-activity-container h2 {
  padding: 0;
  margin: 0;
  line-height: 3.5rem;
  font-size: 2.5rem;
  color: var(--white);
}

.max-ct {
  font-size: 1.5rem;
}

.field-supplies, 
.field-description {
  margin-top: 3rem;
}

#activity-name-input, 
#supply-input, 
#aboutActivity-input {
  border: none;
  border-bottom: 2px solid var(--white);
  width: 80vw;
  background: none;
  color: var(--white);
  height: 50px;
}

#supply-input {
  height: 100px;
}

#aboutActivity-input {
  height: 175px;
}

.add-activity-container textarea {
  resize: none;
  font-size: 1.5rem;
  overflow-y:inherit;
}

.add-activity-container textarea::placeholder {
  font-size: 1.5rem;
  color: var(--white);
}

.add-activity-container button {
  display: inline-block;
	margin-top: 3.5rem;
	margin-right: 1rem;
	padding: .9rem 1rem;
	text-decoration: none;
	font-size: 1.5rem;
	letter-spacing: 1px;
	font-weight: 600;
  border: 2px solid var(--white);
  background: none;
  color: var(--white);
}

.add-activity-container button:hover {
  border: 2px solid white;
  background: white;
  color: var(--deep-rose);
}

@media (min-width:769px) {
  #activity-name-input, 
  #supply-input, 
  #aboutActivity-input {
    max-width: 800px; 
  }
}