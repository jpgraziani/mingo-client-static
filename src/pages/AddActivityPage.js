import React from 'react'

import ApiContext from '../ApiContext'

class AddActivityPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {'supply-input': 'None, just have fun.'}
  }

  handleChange = (event) => {
    this.setState({'supply-input': event.target.value})
  }

  static contextType = ApiContext;

  render() {
    return (
      <article>
        <form action='#'>
          <section className='field'>
            <label htmlFor='activity-name-input'>
              <h4>Activity Name</h4>
            </label>
            <hr />
            <input type='text' id='activity-name-input' name='activity-name-input' required />
          </section>
          <section className='field'>
            <label htmlFor='supply-input'>
              <h4>Supplies</h4>
            </label>
            <hr />
            <textarea type='text' id='supply-input' name='supply-input' value={this.state['supply-input']} onChange={this.handleChange} />
          </section>
          <section className='field'>
            <label htmlFor='aboutActivity-input'>
              <h4>About Activity</h4>
            </label>
            <hr />
            <textarea type='text' id='aboutActivity-input' name='aboutActivity-input'  required/>
          </section>
          <div className='buttons'>
            <button type='submit' className='btns btn-full'>
              Add Activity
            </button>
          </div>
        </form>
      </article>
    );
  }
}

export default AddActivityPage;