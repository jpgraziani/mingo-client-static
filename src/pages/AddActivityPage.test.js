import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddActivityPage from './AddActivityPage'

describe(`AddActivityPage component`, () => {
  it('renders an AddActivityPage by default', () => {
    const wrapper = shallow(<AddActivityPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
