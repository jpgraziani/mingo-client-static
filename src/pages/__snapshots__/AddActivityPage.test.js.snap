// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddActivityPage component renders an AddActivityPage by default 1`] = `
<article
  className="add-activity-container"
>
  <ActivityForm
    className="activity-form"
    onSubmit={[Function]}
  >
    <section
      className="field"
    >
      <label
        htmlFor="activity-name-input"
      >
        <span
          className="max-ct"
        >
          *max 27 characters
        </span>
        <h2>
          Activity Name
        </h2>
      </label>
      <textarea
        id="activity-name-input"
        maxLength="27"
        name="activity-name-input"
        placeholder="enter name"
        required={true}
        type="text"
      />
    </section>
    <section
      className="field-supplies"
    >
      <label
        htmlFor="supply-input"
      >
        <h2>
          Supplies
        </h2>
      </label>
      <textarea
        id="supply-input"
        name="supply-input"
        onChange={[Function]}
        required={true}
        type="text"
        value="None, just have fun."
      />
    </section>
    <section
      className="field-description"
    >
      <label
        htmlFor="aboutActivity-input"
      >
        <h2>
          About Activity
        </h2>
      </label>
      <textarea
        id="aboutActivity-input"
        name="aboutActivity-input"
        placeholder="enter description"
        required={true}
        type="text"
      />
    </section>
    <div
      className="buttons"
    >
      <button
        className="sub-btn"
        type="submit"
      >
        Add Activity
      </button>
    </div>
  </ActivityForm>
</article>
`;
